### PR TITLE
Support tracking branch name for remote git repository (RFE#918)

### PR DESCRIPTION
**Pull request by miurahr**
This allow project manager to maintain their team repository with specific branch name other than 'main' or 'master' default branch.

User can specify branch name when downloading project from dialog.

- [RFE#918](https://sourceforge.net/p/omegat/feature-requests/918/)

TODO:

- [x] field test 
- [x] integration test

Signed-off-by: Hiroshi Miura <miurahr@linux.com>

### CODE REVIEW

File: src/org/omegat/core/team2/impl/GITRemoteRepository2.java [hunk 1]
```diff
@@ -143,6 +158,12 @@ public void init(RepositoryDefinition repo, File dir, ProjectTeamSettings teamSe
             }
             repository = Git.open(localDirectory).getRepository();
             try (Git git = new Git(repository)) {
+                if (trackBranch) {
+                    git.branchCreate().setName(branch).setUpstreamMode(CreateBranchCommand.SetupUpstreamMode.TRACK).setStartPoint("origin/" + branch).call();
```
Comment (miurahr) [inline src/org/omegat/core/team2/impl/GITRemoteRepository2.java:162]: `"origin" + branch` part should be `String.join("/", REMOTE, branch)`

**Comment by amake** [comment]
> This allow project manager to maintain their team repository with specific branch name using "#" for target URL.
> such as "https://github.com/team/repo.git#translate"

This is inventing a new URL format to indicate the git URL + branch. I don't think we should do that.

If there is a standard for such a thing, we should use the standard (I don't think there is).

Otherwise we should either a) allow the user to provide it separately or, preferably, b) automatically detect it.

**Comment by miurahr** [comment]
This has been inspired from [RFE#918](https://sourceforge.net/p/omegat/feature-requests/918/)  and just found in my local working branch today.

>  a) allow the user to provide it separately 

I agree with it.

**Comment by miurahr** [comment]
Because #174 breaks this improvement,  I'd like to update #174 to implement RFE#918.

**Comment by miurahr** [comment]
Superseded by #179

**Pull request closed without merging**